internal/manifests/collector/configmap.go

```diff
@@ -32,5 +39,49 @@ func ConfigMap(params manifests.Params) (*corev1.ConfigMap, error) {
 		Data: map[string]string{
 			"cwagentconfig.json": replacedConf,
 		},
-	}, nil
+	})
+
+	if !params.OtelCol.Spec.Prometheus.IsEmpty() {
+		promName := naming.PrometheusConfigMap(params.OtelCol.Name)
+		promLabels := manifestutils.Labels(params.OtelCol.ObjectMeta, promName, "", ComponentAmazonCloudWatchAgent, []string{})
+
+		replacedPrometheusConf, err := ReplacePrometheusConfig(params.OtelCol)
+		if err != nil {
+			params.Log.V(2).Info("failed to update prometheus config to use sharded targets: ", "err", err)
+			return nil, err
+		}
+
+		if !params.OtelCol.Spec.TargetAllocator.Enabled {
+			replacedPrometheusConfig, err := adapters.ConfigFromString(replacedPrometheusConf)
+			if err != nil {
+				return nil, err
+			}
+
+			replacedPrometheusConfProp, ok := replacedPrometheusConfig["config"]
+			if !ok {
+				return nil, fmt.Errorf("no prometheusConfig available as part of the configuration")
+			}
+
+			replacedPrometheusConfPropYAML, err := yaml.Marshal(replacedPrometheusConfProp)
+			if err != nil {
+				return nil, err
+			}
+
+			replacedPrometheusConf = string(replacedPrometheusConfPropYAML)
+		}
+
+		configmaps = append(configmaps, &corev1.ConfigMap{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:        promName,
+				Namespace:   params.OtelCol.Namespace,
+				Labels:      promLabels,
+				Annotations: params.OtelCol.Annotations,
+			},
+			Data: map[string]string{
+				"prometheus.yaml": replacedPrometheusConf,
```

Comment:
Should `"prometheus.yaml"` be a constant or `ConfigMapEntry()`?

Reply:
lets change it to the ConfigMapEntry()